### ISSUE TITLE
Python Console fails to open if QGIS is build without 3D support (like for MinGW builds)

### ISSUE THREAD

**Issue by agiudiceandrea** [qgis/QGIS#41855]
**Describe the bug**

~~In the MinGW build provided in the continuous integrations process, it is not possible to use the Python Console.~~
The Python Console fails to open if QGIS is build without 3D support like for MinGW builds provided in the continuous integrations process.
The following error is displayed trying to open it:

```
Failed to open Python console:

Traceback (most recent call last):
  File "F:\QGIS\QGIS-Portable\lib\python3.9\code.py", line 90, in runcode
    exec(code, self.locals)
  File "", line 1, in 
  File "F:\QGIS\QGIS-Portable\lib\python3.9\site-packages\qgis\utils.py", line 798, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
ModuleNotFoundError: No module named 'qgis._3d'
...
```
See the log [mingw_python_console.txt](https://github.com/qgis/QGIS/files/6048115/mingw_python_console.txt)

As a workaround, removing the `"from qgis._3d import *"` from `_init_commands` variable in [console_sci.py](https://github.com/qgis/QGIS/blob/master/python/console/console_sci.py) fixes the issue and it is possible to open and use the Python Console.

~~It seem QGIS was not built using the QGIS 3D library: so to permanently fix the issue QGIS should be built using the QGIS 3D library, if possible, or~~ So it would be better to set an exception in console_sci.py in order to avoid to try to import qgis._3d if QGIS was not built using the QGIS 3D library.

**QGIS and OS versions**

QGIS version:
3.19.0-Master 'Master', 7b37f32b

Windows 7 SP 1 64 bit

**Additional context**

The default import of the 3D python library in console was introduced with PR https://github.com/qgis/QGIS/pull/36784 (https://github.com/qgis/QGIS/commit/601a71717407af3a1f6a544c65a9244074824662) in order to fix https://github.com/qgis/QGIS/issues/36661

**Comment by elpaso** [comment]
Actually, since it is possible to build QGIS without 3d, I think that the bug is in the console code trying to import a module that was not built.

**Comment by agiudiceandrea** [comment]
Yes you are right! I will modify this issue report.
What do you think about https://github.com/qgis/QGIS/issues/41857?

**Comment by agiudiceandrea** [comment]
It seems to me the issue is not yet fixed and still occurs also after #42634:

```
Failed to open Python console:


Traceback (most recent call last):
  File "F:\QGIS\QGIS-Portable\lib\python3.9\code.py", line 90, in runcode
    exec(code, self.locals)
  File "", line 1, in 
  File "F:\QGIS\QGIS-Portable\lib\python3.9\site-packages\qgis\utils.py", line 799, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
ModuleNotFoundError: No module named 'qgis._3d'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "", line 2, in 
  File "F:/QGIS/QGIS-Portable/share/qgis/python\console\console.py", line 57, in show_console
    _console = PythonConsole(parent)
  File "F:/QGIS/QGIS-Portable/share/qgis/python\console\console.py", line 110, in __init__
    self.console = PythonConsoleWidget(self)
  File "F:/QGIS/QGIS-Portable/share/qgis/python\console\console.py", line 141, in __init__
    self.shell = ShellScintilla(self)
  File "F:/QGIS/QGIS-Portable/share/qgis/python\console\console_sci.py", line 71, in __init__
    self.runsource(line)
  File "F:/QGIS/QGIS-Portable/share/qgis/python\console\console_sci.py", line 555, in runsource
    return super(ShellScintilla, self).runsource(source, filename, symbol)
  File "F:\QGIS\QGIS-Portable\lib\python3.9\code.py", line 74, in runsource
    self.runcode(code)
  File "F:\QGIS\QGIS-Portable\lib\python3.9\code.py", line 94, in runcode
    self.showtraceback()
  File "F:\QGIS\QGIS-Portable\lib\python3.9\code.py", line 148, in showtraceback
    sys.excepthook(ei[0], ei[1], last_tb)
  File "F:/QGIS/QGIS-Portable/share/qgis/python\console\console_sci.py", line 551, in excepthook
    self.write("".join(traceback.format_exception(etype, value, tb)))
  File "F:/QGIS/QGIS-Portable/share/qgis/python\console\console_sci.py", line 536, in write
    sys.stderr.write(txt)
AttributeError: 'NoneType' object has no attribute 'write'
```

**Comment by agiudiceandrea** [comment]
@gioman could you please reopen this issue report?
Despite of the PR #42634 merged in master by @nyalldawson, the issue keep occurring in the latest QGIS 3.19-master 9655b710.
